examples/ft-refund/solidity-contract/src/FtRefund.sol

```diff
@@ -0,0 +1,194 @@
+        );
+        approveCall.transact();
+    }
+
+    // Given an ERC-20 token that was bridged to Aurora from Near,
+    // this function will call the `ft_transfer_call` function on the
+    // Near-native NEP-141 token that ERC-20 was bridged from.
+    // This would be an important piece of an XCC-based integration into
+    // a Near app such as ref.finance (allowing Aurora users to use the
+    // Near app without creating a Near account themselves).
+    // For the purpose of this example, the call includes a message to
+    // force a refund back to the user because the point of this example
+    // is to show how to refund the Aurora user in the case there is a failure
+    // on the Near side.
+    function ftTransferCall(
+        IEvmErc20 token,
+        string memory tokenId,
+        uint128 amount
+    ) public {
+        token.transferFrom(msg.sender, address(this), amount);
+        token.withdrawToNear(
+            abi.encodePacked(AuroraSdk.nearRepresentative(address(this))),
+            uint(amount)
+        );
+
+        bytes memory data = abi.encodePacked(
+            "{",
+            '"receiver_id": "',
+            nearAccountId,
+            '",',
+            '"amount": "',
+            Strings.toString(amount),
+            '",',
+            '"msg": "refund"',
+            "}"
+        );
+        PromiseCreateArgs memory callFtTransfer = near.call(
+            tokenId,
+            "ft_transfer_call",
+            data,
+            1,
+            FT_TRANSFER_CALL_NEAR_GAS
+        );
+        PromiseCreateArgs memory callback = near.auroraCall(
+            address(this),
+            abi.encodeWithSelector(
+                this.ftTransferCallCallback.selector,
+                msg.sender,
+                tokenId,
+                amount
+            ),
+            0,
+            CALLBACK_NEAR_GAS
+        );
+
+        callFtTransfer.then(callback).transact();
+    }
+
+    // This function is the callback from `ftTransferCall` which refunds the user
+    // in the case that `ft_transfer_call` did not comsume all the tokens the user
+    // sent. This is accomplished by bridging the tokens returned to the XCC account
+    // on Near back to Aurora via another `ft_transfer_call`.
+    function ftTransferCallCallback(
+        address sender,
+        string memory tokenIdOnNear,
+        uint128 amount
+    ) public onlyRole(CALLBACK_ROLE) {
+        PromiseResult memory promiseResult = AuroraSdk.promiseResult(0);
+        uint128 refundAmount = 0;
+
+        if (promiseResult.status != PromiseResultStatus.Successful) {
+            // if Promise failed we need to do whole refund
+            refundAmount = amount;
+        } else {
+            // else `ft_on_transfer` will return used amount of FT,
+            // which we need to extract from original amount
+            uint128 usedAmount = _stringToUint(string(promiseResult.output));
+            refundAmount = amount - usedAmount;
```

Comment:
According to [spec](https://nomicon.io/Standards/Tokens/FungibleToken/Core#reference-level-explanation) `ft_on_transfer` should return the number of _unused_ tokens. Accordingly `near-contract::ft_on_transfer` returns `amount` if `&msg == "refund"`, i.e. the whole amount is unused. Then the interpretation of `promiseResult.output` here is wrong, isn't it?